state-chain/pallets/cf-ingress-egress/src/tests.rs

```diff
@@ -2951,4 +2977,72 @@ mod evm_transaction_rejection {
 				assert!(deposit_ids.contains(&&TAINTED_TX_ID_2));
 			});
 	}
+
+	#[test]
+	fn rejecting_vault_swap() {
+		let tx_id =
+			H256::from_str("0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef")
+				.unwrap();
+
+		new_test_ext()
+			.then_execute_with(|_| {
+				assert_ok!(IngressEgress::mark_transaction_for_rejection(
+					OriginTrait::signed(BROKER),
+					tx_id,
+				));
+			})
+			.then_execute_at_next_block(|_| {
+				assert_ok!(submit_vault_swap_request(
+					Asset::Eth,
+					Asset::Flip,
+					DEFAULT_DEPOSIT_AMOUNT,
+					Default::default(),
+					MockAddressConverter::to_encoded_address(ForeignChainAddress::Eth(
+						ALICE_ETH_ADDRESS
+					)),
+					None,
+					tx_id,
+					DepositDetails { tx_hashes: Some(vec![tx_id]) },
+					Beneficiary { account: BROKER, bps: 0 },
+					bounded_vec![],
+					ETH_REFUND_PARAMS,
+					None,
+					0
+				));
+				assert!(MockSwapRequestHandler::<Test>::get_swap_requests().is_empty());
+
+				let scheduled_txs = ScheduledTransactionsForRejection::<Test, ()>::get();
+
+				assert_eq!(scheduled_txs.len(), 1);
+				assert_eq!(scheduled_txs[0].deposit_details.deposit_ids().unwrap(), vec![tx_id]);
+			})
+			.then_process_blocks(1)
+			.then_process_events(|_, event| match event {
+				RuntimeEvent::IngressEgress(PalletEvent::TransactionRejectedByBroker {
+					tx_id,
+					..
+				}) if tx_id == tx_id => Some(()),
+				_ => {
+					panic!("Expected a TransactionRejectedByBroker event");
```

Comment:
strictly speaking this will panic if there is any event that is not TransactionRejectedByBroker.

(`"Unexpected event"`)

In order to check if there were any `TransactionRejectedByBroker` events, you need to read the vec that is passed through to the next closure.